Mobile/ShopLens/Droid/IO/ShopLensPictureDirectoryCreator.cs

```diff
@@ -14,7 +14,10 @@ private void CreateDirectoryForPictures(File pictureDirectory)
         {
             if (!pictureDirectory.Exists())
             {
-                pictureDirectory.Mkdirs();
+                if (!pictureDirectory.Mkdirs())
+                {
+                    throw new System.IO.DirectoryNotFoundException("Unable to create directory for pictures.");  
```

Comment:
Are you sure it is DirectoryNotFoundException when the comment actually says "Unable to create directory for pictures."?

Reply:
Well, System.IO doesn't have an exception that refers to directories except this one, so we might need to create our own exception class. Don't know if it's worth it.

Reply:
But isn't this a "Permission Denied" exception? https://developer.android.com/reference/java/lang/SecurityException

Reply:
Huh, I thought that we shouldn't throw Java exceptions because the directory creator shouldn't be attached to the Xamarin platform, but since the method Mkdirs() is Xamarin specific and it isn't Directory.CreateDirectory(), I guess we should throw Java exceptions. maybe we should change the name of the class then?

Reply:
Call it ShopLensPictureDirectoryCreatorXamarin or is that redundant?

Reply:
> the directory creator shouldn't be attached to the Xamarin platform

What do you mean? We're already attached to the Xamarin platform. If we didn't want to be attached, we would have used Java 😄

Reply:
Well, I suppose. But Directory Creator doesn't need to use Java, but then it wouldn't be able to create directories on an Android device?

Reply:
I mean, Directory.CreateDirectory() wouldn't do anything.

Reply:
I thought that you could use Directory.CreateDirectory() on an Android device.

Reply:
Oh wait, you can use Directory.CreateDirectory() on an Android device, so my point stands.

Reply:
I was just thinking that Directory Creator wouldn't work on a non Xamarin project, which doesn't make it a directory creator, it makes it a Xamarin directory creator or a Java directory creator.

Reply:
But I guess ShopLens is on Xamarin, so it can be just a ShopLens directory creator?